src/validation/multicityValidation.js

```diff
@@ -0,0 +1,124 @@
+import { body, validationResult } from 'express-validator';
+import { Op } from 'sequelize';
+import models from '../models';
+
+const { Locations } = models;
+
+const multicityCheck = [
+  body('departureDate').trim().not().isEmpty()
+    .withMessage('Departure Date field is required')
+    .matches(/^\d{4}([-./,:])\d{2}\1\d{2}$/, 'i')
+    .withMessage('The date must follow date format YYYY-MM-DD'),
+  body('returnDate').trim().not().isEmpty()
+    .withMessage('Return Date field is required')
+    .matches(/^\d{4}([-./,:])\d{2}\1\d{2}$/, 'i')
+    .withMessage('The date must follow date format YYYY-MM-DD'),
+  body('currentOfficeLocation', 'current Office Location field is required').trim().not().isEmpty(),
+  body('reason').trim().not().isEmpty()
+    .withMessage('Reason field is required'),
+  body('tripType').trim().not().isEmpty()
+    .withMessage('The trip type field is required')
+    .matches(/^Multi-city$/, 'i')
+    .withMessage('The trip type must match the type "Multi-city"')
+];
+
+const validateOrigin = async (request) => {
+  const errors = {};
+  const { currentOfficeLocation } = request.body;
+  if (!currentOfficeLocation) {
+    if (!errors.currentOfficeLocation) errors.currentOfficeLocation = ['currentOfficeLocation is required'];
+    else errors.currentOfficeLocation.push('currentOfficeLocation is required');
+  }
+  if (!/^\d+$/.test(currentOfficeLocation)) {
+    if (!errors.currentOfficeLocation) errors.currentOfficeLocation = ['currentOfficeLocation must be a number'];
+    else errors.currentOfficeLocation.push('currentOfficeLocation must be a number');
+  }
+  if (!errors.currentOfficeLocation && currentOfficeLocation) {
+    const locationsData = await Locations.findAll({
+      attributes: ['id', 'locationName'],
+      where: { id: currentOfficeLocation },
+      raw: true
+    });
+    if (!locationsData.length) {
+      if (!errors.currentOfficeLocation) errors.currentOfficeLocation = ['Current office location does not exist'];
+      else errors.currentOfficeLocation.push('Current office location does not exist');
+    }
+    if (locationsData.length) {
+      const currentOfficeData = {};
+      currentOfficeData[locationsData[0].locationName] = locationsData[0].id;
+      request.currentOfficeData = currentOfficeData;
+    }
+  }
+  return errors;
+};
+const validateDestination = async (request) => {
+  const errors = {};
+  const { destination } = request.body;
+  if (!destination) {
+    if (!errors.destination) errors.destination = ['Destination input be an array', 'Mutiple office locations inputs are required'];
+    else errors.destination.push('Destination input be an array', 'Mutiple office locations inputs are required');
```

Comment:
The error message here should be something like `Destination input must be an array` not `Destination input be an array`